test/test-tcp-reuseport.c

```diff
@@ -157,22 +158,36 @@ TEST_IMPL(tcp_reuseport) {
 
   r = uv_mutex_init(&mutex);
 
-  /* Create listener per event loop. */
   main_loop = uv_default_loop();
   ASSERT_NOT_NULL(main_loop);
-  create_listener(main_loop, &main_handle);
-  uv_timer_init(main_loop, &main_timer_handle);
-  uv_timer_start(&main_timer_handle, ticktack, 0, 10);
 
-  thread_loop = uv_loop_new();
-  ASSERT_NOT_NULL(thread_loop);
-  create_listener(thread_loop, &thread_handle);
-  uv_timer_init(thread_loop, &thread_timer_handle);
-  uv_timer_start(&thread_timer_handle, ticktack, 0, 10);
+  /* Create listener per event loop. */
 
-  /* Connect to the peers. */
+  thread_loop1 = uv_loop_new();
+  ASSERT_NOT_NULL(thread_loop1);
+  create_listener(thread_loop1, &thread_handle1);
+  uv_timer_init(thread_loop1, &thread_timer_handle1);
+  uv_timer_start(&thread_timer_handle1, ticktack, 0, 10);
+
+  thread_loop2 = uv_loop_new();
+  ASSERT_NOT_NULL(thread_loop2);
+  create_listener(thread_loop2, &thread_handle2);
+  uv_timer_init(thread_loop2, &thread_timer_handle2);
+  uv_timer_start(&thread_timer_handle2, ticktack, 0, 10);
+
+  /* Run event loops of listeners in separate threads. */
+  uv_thread_t thread_loop_id1;
+  uv_thread_t thread_loop_id2;
+  uv_thread_create(&thread_loop_id1, run_event_loop, thread_loop1);
+  uv_thread_create(&thread_loop_id2, run_event_loop, thread_loop2);
+
+  /* Sleep to ensure all threads to poll for accepting connections
+   * before we start to connect. Otherwise the incoming connections
+   * might not be distributed across all listening threads. */
+  uv_sleep(100);
```

Comment:
Perhaps you could use a semaphore to make sure rather than relying on timing?

Reply:
Good point, done.